shared/middleware/helpers/shared_auth_helpers.rb

```diff
@@ -0,0 +1,27 @@
+require 'cdo/user_helpers'
+require 'json'
+
+#
+# Utility methods that help middleware access dashboard authentication and
+# permissions information.
+#
+# The use of these methods within shared and pegasus is deprecated.
+# These methods should be eliminated or moved to dashboard.
+#
+
+# @returns [Integer] the user_id associated with the current request
+def current_user_id
+  # @request is a Sinatra::Request < Rack::Request provided by Sinatra::Base
+  @request.nil? ? nil : @request.user_id
+end
+
+# @returns [Hash] the dashboard user row associated with the current request.
+def current_user
+  return nil if (id = current_user_id).nil?
+  @dashboard_user ||= DASHBOARD_DB[:users][id: id]
+end
+
+# @returns [Boolean] true if the current user is an admin.
+def admin?
+  current_user && !!current_user[:admin]
+end
```

Comment:
still used outside of dashboard here: https://github.com/code-dot-org/code-dot-org/blob/6e49521fd97dd1e7ced61f801863ad3b48a0d71e/pegasus/helper_modules/dashboard.rb#L65-L64

Reply:
`has_permission?` just got removed in https://github.com/code-dot-org/code-dot-org/pull/47921, so now `admin?` can move to dashboard too.